hutch_python/plugins/beamline.py

```diff
@@ -0,0 +1,32 @@
+from pcdsdevices.daq import Daq, make_daq_run_engine
+from pcdsdevices.sim.daq import SimDaq
+
+from ..base_plugin import BasePlugin
+from .happi import Plugin as HappiPlugin
+
+DAQ_MAP = dict(mfx=4)
+HAPPI_DB = 'filename add here later'
+SIM_DAQ = False
+
+
+class Plugin(BasePlugin):
+    name = 'beamline'
+
+    def pre_plugins(self):
+        requirements = dict(active=True,
+                            beamline=self.info.upper())
+        info = dict(filename=HAPPI_DB,
+                    requirements=requirements)
+        return [HappiPlugin(info=info)]
+
+    def get_objects(self):
+        # Make the Daq object
+        hutch = self.info.lower()
+        if SIM_DAQ:
+            Cls = SimDaq
+        else:
+            Cls = Daq
+        daq = Cls(name='daq', platform=DAQ_MAP[hutch])
+        # Make the Daq RunEngine
+        RE = make_daq_run_engine(daq)
+        return dict(daq=daq, RE=RE)
```

Comment:
How does this work in the current hutch python? Does every session connect to pydaq? Can you connect from machines where the DAQ is not running?

Reply:
You can import `pydaq` without connecting to the DAQ, but you have to avoid using it. I was thinking we'd work toward either a command-line option to swap to `SimDaq` instead of `Daq` for these other machines (helpful for debugging), or we can try to dynamically decide which machines should load the real daq... Actually, this isn't quite good enough, since what we *really* want is a way to have this return just a vanilla RE for non-daq work.